frontend/app/chat/[chatId]/hooks/useChat.ts

```diff
@@ -21,30 +22,34 @@ export const useChat = () => {
   const {
     config: { maxTokens, model, temperature },
   } = useBrainConfig();
-  const { history, setHistory, addToHistory } = useChatContext();
+  const { history, setHistory } = useChatContext();
   const { publish } = useToast();
 
   const {
     createChat,
     getChatHistory,
-    addQuestion: addQuestionToChat,
+    addStreamQuestion,
+    addQuestion: addQuestionToModel,
   } = useChatService();
 
   useEffect(() => {
     const fetchHistory = async () => {
-      const chatHistory = await getChatHistory(chatId);
-      setHistory(chatHistory);
+      const currentChatId = chatId;
+      const chatHistory = await getChatHistory(currentChatId);
+
+      if (chatId === currentChatId && chatHistory.length > 0) {
```

Comment:
Not sure to understand what is happening here:
```
const currentChatId = chatId;
if(chatId === currentChatId)
````

Reply:
fair :) that's dead. I think I didnt spot the line above.